crates/faucet/src/lib.rs

```diff
@@ -230,12 +236,32 @@ impl Faucet {
     /// Creates a transaction with the given notes, executes it, proves it, and submits using the
     /// local miden-client. This results in submitting the transaction to the node and updating the
     /// local db to track the created notes.
-    async fn create_transaction(
-        &mut self,
-        notes: Vec<OutputNote>,
-    ) -> Result<TransactionId, ClientError> {
+    async fn create_transaction(&mut self, notes: Vec<Note>) -> Result<TransactionId, ClientError> {
         // Build the transaction
-        let tx = TransactionRequestBuilder::new().own_output_notes(notes).build()?;
+        let expected_output_recipients = notes.iter().map(Note::recipient).cloned().collect();
+        let n = notes.len() as u64;
+        let mut note_data = vec![Felt::new(n)];
+        for note in notes {
+            // SAFETY: these are p2id notes with only one fungible asset
+            let amount = note.assets().iter().next().unwrap().unwrap_fungible().amount();
+
+            note_data.push(note.recipient().digest()[0]);
+            note_data.push(note.recipient().digest()[1]);
+            note_data.push(note.recipient().digest()[2]);
+            note_data.push(note.recipient().digest()[3]);
```

Comment:
Would something like `note_data.extend(recipient().digest().iter())` not work here?